src/App.tsx

```diff
@@ -1,26 +1,279 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
+
 import React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { TempTodoItem } from './components/TempTodoItem';
+import { showError } from './utils/showError';
+import { TodoHeader } from './components/TodoHeader';
+import { ListOfTodos } from './components/ListOfTodos';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAdding, setIsAdding] = useState(false);
+  const [isDelete, setIsDelete] = useState<number | null>(null);
+  const [updatingTodos, setUpdatingTodos] = useState<{
+    [key: number]: boolean;
+  }>({});
```

Comment:
```suggestion
  const [updatingTodos, setUpdatingTodos] = useState<{
    [key: number]: boolean;
  }>({});
```
only update as i remember you need such state and for delete
so processingIds naming suits better